src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -115,6 +115,13 @@ public static void DictionaryOfObject()
             Assert.Equal(@"{""Key1"":1}", json);
         }
 
+        [Fact]
+        public static void EmptyDictionaryOfObject()
+        {
+            Assert.Throws<JsonException>(() => JsonSerializer.Parse<Dictionary<string, string>>("\"headers\""));
```

Comment:
What happens if the type being deserialized is a POCO with a dictionary property?

Reply:
I tried to parse this and its working perfectly
```
Poco2 t = JsonSerializer.Parse<Poco2>("{\n\t\"ani\" : {\"a\" : \"b\", \"c\" : \"d\"}}");
Assert.Equal(t.ani["a"], "b");
Assert.Equal(t.ani["c"], "d");

class Poco2
{
        public Dictionary<string, string> ani  { get; set; }
}
```

Reply:
I mean in the invalid case, where the payload is just `"\"headers\""`

So, in your case, what does the following do? Does it throw? We should verify and add to the test case.
```C#
JsonSerializer.Parse<Poco2>("\"headers\"");
```